lib/app.js

```diff
@@ -0,0 +1,771 @@
+        // In dev environments, route the individual files.
+        if (self.isDev) {
+          load.route();
+        }
+        loads[url] = load.concat(url).route();
+        self.logWhenLoadIsFinished(loads[url], dir, url);
+      }
+    });
+
+    chug.onceReady(function () {
+
+      [self.models, self.controllers].forEach(function (collection) {
+        collection.each(function callReady(asset) {
+          var instance = asset.module.instance;
+          instance.setFlag('ready');
+        });
+      });
+
+      self.views.assets.forEach(function (asset) {
+        var name = asset.location.replace(/(^.*[\/\\]views[\/\\]|\.[a-z]+$)/g, '');
+        self.views[name] = asset;
+      });
+
+      // In dev and debug environments, watch for changes.
+      if (self.isDev) {
+        var watchStream = process.stdin;
+        JSON.readStream(watchStream, 'event');
+        watchStream.on('event', function (event) {
+          if (event.type == 'change') {
+            var path = event.path;
+            var asset = chug.cache.get(path);
+            self.logger.warn('[Lighter] Changed ' + shorten(path).cyan + '.');
+            if (asset) {
+              asset.parents.forEach(function (load) {
+                load.handleChange(path);
+              });
+              self.beams.emit('chug:change', path);
+            }
+            else {
+              process.exit();
+            }
+          }
+        });
+      }
+
+      // In stage and prod environments, minify assets.
+      if (self.isProd) {
+        self.logger.warn("[Lighter] Minifying assets... ");
+        var wait = 1;
+        var unwait = function () {
+          if (!--wait) {
+            self.logger.info("[Lighter] Views, scripts and styles minified.");
+            self.setFlag('loaded');
+          }
+        };
+        [self.scripts, self.styles].forEach(function (loads) {
+          for (var url in loads) {
+            wait++;
+            loads[url].wrap().minify().gzip().route().then(unwait);
+          }
+        });
+        self.views.minify().then(unwait);
+      }
+      // Otherwise, just signal that they're already loaded.
+      else {
+        self.setFlag('loaded');
+      }
+
+    });
+  },
+
+  initDbs: function () {
+    var self = this;
+    var config = self.config;
+    self.orm = require(config.orm);
+    self.dbs = {};
+    config.dbs = config.dbs || {db: config.db};
+    for (var dbKey in config.dbs) {
+      var dbConfig = config.dbs[dbKey];
+      dbConfig.name = dbConfig.name || dbKey;
+      dbConfig.app = self;
+      dbConfig.logger = self.logger;
+      var db = self.orm(dbConfig, self);
+      self.dbs[dbKey] = db;
+
+      // Expose the first (and possibly the only) database as "db".
+      if (!self.db) {
+        self.db = db;
+      }
+    }
+
+    // Allow databases to be exposed globally.
+    if (config.exposeGlobals) {
+      global.Db = self.db;
+      global.Dbs = self.dbs;
+    }
+  },
+
+  initRings: function () {
```

Comment:
is this for ringer? not sure if it would be a gd idea to be included in, may be leave this snippet out for now?